visualization/app/codeCharta/ui/codeMap/codeMap.arrow.service.spec.ts

```diff
@@ -88,12 +98,33 @@ describe("CodeMapArrowService", () => {
 			expect(CodeMapMouseEventService.subscribeToBuildingUnhovered).toHaveBeenCalledWith($rootScope, codeMapArrowService)
 		})
 	})
+	describe("Arrow Behaviour when selecting and hovering a building", () => {
+		it("should only highlight small leaf when big leaf is selected", () => {
+			storeService.dispatch(setEdges(VALID_EDGES_DECORATED))
+			const nodes: Node[] = [
+				CODE_MAP_BUILDING_WITH_OUTGOING_EDGE_NODE.node,
+				CODE_MAP_BUILDING_WITH_INCOMING_EDGE_NODE.node,
+				DIFFERENT_NODE
+			]
+			const edges: Edge[] = storeService.getState().fileSettings.edges.filter(x => x.visible !== EdgeVisibility.none)
```

Comment:
Nit (the type should be inferred):

```suggestion
			const edges = storeService.getState().fileSettings.edges.filter(x => x.visible !== EdgeVisibility.none)
```